x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/per_field_rule_diff_tab.test.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  KqlQueryType,
+  ThreeWayDiffOutcome,
+  ThreeWayMergeOutcome,
+} from '../../../../../common/api/detection_engine';
+import type { PartialRuleDiff } from '../../../../../common/api/detection_engine';
+import { TestProviders } from '../../../../common/mock';
+import { render } from '@testing-library/react';
+import React from 'react';
+import { PerFieldRuleDiffTab } from './per_field_rule_diff_tab';
+
+const ruleFieldsDiffBaseFieldsMock = {
+  diff_outcome: ThreeWayDiffOutcome.StockValueCanUpdate,
+  has_conflict: false,
+  has_update: true,
+  merge_outcome: ThreeWayMergeOutcome.Target,
+};
+
+const ruleFieldsDiffMock: PartialRuleDiff = {
+  fields: {
+    version: {
+      ...ruleFieldsDiffBaseFieldsMock,
+      base_version: 1,
+      current_version: 1,
+      merged_version: 2,
+      target_version: 2,
+    },
+  },
+  has_conflict: false,
+};
+
+const renderPerFieldRuleDiffTab = (ruleDiff: PartialRuleDiff) => {
+  return render(
+    <TestProviders>
+      <PerFieldRuleDiffTab ruleDiff={ruleDiff} />
+    </TestProviders>
+  );
+};
+
+describe('PerFieldRuleDiffTab', () => {
+  test('Field groupings should be rendered together in the same accordion panel', () => {
+    const mockData: PartialRuleDiff = {
+      ...ruleFieldsDiffMock,
+      fields: {
+        kql_query: {
+          ...ruleFieldsDiffBaseFieldsMock,
+          base_version: {
+            filters: [],
+            language: 'lucene',
+            query: 'old query',
+            type: KqlQueryType.inline_query,
+          },
+          current_version: {
+            filters: [],
+            language: 'lucene',
+            query: 'old query',
+            type: KqlQueryType.inline_query,
+          },
+          merged_version: {
+            filters: [],
+            language: 'kuery',
+            query: 'new query',
+            type: KqlQueryType.inline_query,
+          },
+          target_version: {
+            filters: [],
+            language: 'kuery',
+            query: 'new query',
+            type: KqlQueryType.inline_query,
+          },
+        },
+      },
+    };
+    const wrapper = renderPerFieldRuleDiffTab(mockData);
+
+    const matchedSubtitleElements = wrapper.queryAllByTestId('ruleUpgradePerFieldDiffSubtitle');
+    const subtitles = matchedSubtitleElements.map((element) => {
+      return element.textContent;
+    });
+
+    // `filters` and `type` have not changed between versions so shouldn't be displayed
+    expect(subtitles).toEqual(['Query', 'Language']);
+  });
+
+  describe('Undefined values are displayed with empty diffs', () => {
+    test('when a new field is added', () => {
```

Comment:
Usually `describe` defines a context and `test`/`it` says what it tests. It looks inverced here and makes tests reading more complicated. Two `test` sections would work better here named like

- displays only an updated field value when changed from undefined
- displays only an outdated field value when incoming update is undefined